### ISSUE TITLE
Error message given by toHaskellFunction

### ISSUE THREAD

**Issue by hth313** [osa1/hslua#59]
When an error occurs it prepends the error message with `Error while calling haskell function: `.

Talking about "Haskell" may be confusing and misleading to a user of an application that internally uses hslua. The user is not using Haskell, (s)he is using an application and the error is related to that application, not Haskell.

Would it be possible to just remove this prefix text and trust that the reported error is clear enough, or would it be better to have alternative functions, which could allow for specifying the prefixing text (possibly leaving it empty)?

A minor issue is that "haskell" should probably be "Haskell" in the prefix string.

**Comment by tarleb** [comment]
I agree, it's probably better to change this. I hadn't considered that this might be confusing to users.

There is probably a better way to do error handling than we do right now. Usually, there is an error happening in lua, and the calling functions need to know about the context in which the error occured. Handling this context by prefixing strings is probably not the best method to handle this context.

Anyway, I'll remove "haskell" from the error message and release a new minor version later this week.

**Comment by tarleb** [comment]
Wht do you think returning a table instead of a simple string in case of an error? We could encode more details on the error while still providing a standard way to generate an error string by adding a `__tostring` metamethod. It would make it much easier to change the error message without having to resort to substring substitution. The disadvantage I see is that one would always have to call `tostring` if a string message is required, but this might be worth the added flexibility.

**Comment by hth313** [comment]
I think a simple message should be enough until proven wrong.

**Comment by tarleb** [comment]
The new version has been uploaded. I didn't find the time to do some other changes I had in mind, so the altered error message is the only change in that release.
